cmd/sql-main.go

```diff
@@ -68,11 +129,220 @@ EXAMPLES:
 
    3. Run a query on an encrypted object with customer provided keys.
       $ {{.HelpName}} --encrypt-key "myminio/iot-devices=32byteslongsecretkeymustbegiven1" \
-            --query "select count(s.power) from S3Object" myminio/iot-devices/power-ratio-encrypted.csv
+		      --query "select count(s.power) from S3Object s" myminio/iot-devices/power-ratio-encrypted.csv
+	 
+   4. Run a query on an object on Minio in gzip format using ; as field delimiter,
+      newline as record delimiter and file header to be used 
+      $ {{.HelpName}} --compression GZIP --csv-input "rd=\n,fh=USE,fd=;" \
+		      --query "select count(s.power) from S3Object" myminio/iot-devices/power-ratio.csv.gz
+	 
+   5. Run a query on an object on Minio in gzip format using ; as field delimiter,
+      newline as record delimiter and file header to be used 
+      $ {{.HelpName}} --compression GZIP --csv-input "rd=\n,fh=USE,fd=;" \ 
+               --json-output "rd=\n\n" --query "select * from S3Object" myminio/iot-devices/data.csv
+
 `,
 }
 
-func sqlSelect(targetURL, expression string, encKeyDB map[string][]prefixSSEPair) *probe.Error {
+// filter json from allowed flags for sql command
+func getSQLFlags() []cli.Flag {
+	flags := append(sqlFlags, ioFlags...)
+	for _, f := range globalFlags {
+		if f.GetName() != "json" {
+			flags = append(flags, f)
+		}
+	}
+	return flags
+}
+
+// valid CSV and JSON keys for input/output serialization
```

Comment:
can you put this in a single var block ? 

```
var (
)
```

Reply:
done